lib/utils/timetable.ts

```diff
@@ -116,26 +117,41 @@ export const toDefaultLocalDateString = (dateString: string) => {
   });
 };
 
-type StageProgress = "IN PROGRESS" | "FINISHED" | "NOT STARTED";
+const getDatePartFromISOString = (isoString: string, includeDays: boolean) => {
+  const date = isoString.split("T")[0];
+  return includeDays ? date : date.split("-").slice(0, 2).join("-");
+};
 
 export const getStageProgress = (
   lastUpdatedDate: string,
   startEventDate: string,
   endEventDate?: string
-): StageProgress => {
-  const startDate = new Date(startEventDate);
-  const referenceDate = new Date(lastUpdatedDate);
+): TagLabel => {
+  const includeDays = startEventDate.split("-").length > 2;
+  const startDate = new Date(
+    getDatePartFromISOString(startEventDate, includeDays)
```

Comment:
what prompted you to use getDatePartFromISOString here?

Reply:
I am not sure I understand why we aren't just doing: new Date(startEventDate)

Reply:
It does thanks :). I understand why we want to strip the time and, eventually, the day from lastUpdatedDate. But why are we doing it for startDate? is that in case startDate also comes in as a full ISOString? also why are we not doing the same for the endDate on line 137?

Reply:
Ahh, that's a very good question 😆 I did it for start date as a "just in case" even though it wouldn't ever really be needed (at the moment, at least). I didn't do it for the end date because I forgot...

Reply:
I've added the formatting to the end date as well